sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py

```diff
@@ -427,6 +436,55 @@ def _update_inputs(self, pipeline_inputs):
                 anno.description = self._args_description.get(input_name)
                 self.inputs[input_name] = anno
 
+    @classmethod
+    def _get_output_annotation(cls, func):
+        """Get the output annotation of the function, validate & refine it."""
+        return_annotation = inspect.signature(func).return_annotation
+        # skip if return annotation is not group
+        if not is_group(return_annotation):
+            return {}
+        outputs = _get_param_with_standard_annotation(return_annotation, is_func=False)
+        output_annotations = {}
+        for key, val in outputs.items():
+            if isinstance(val, GroupInput):
+                raise UserErrorException(message="Nested group annotation is not supported in pipeline output.")
+            # normalize annotation since currently annotation in @group will be converted to Input
+            if isinstance(val, Input):
+                val = Output(type=val.type)
+            if not isinstance(val, Output):
+                raise UserErrorException(
+                    message="Invalid output annotation. "
+                            f"Only Output annotation in return annotation is supported. Got {type(val)}."
+                )
+            output_annotations[key] = val._to_dict()
+        return output_annotations
+
+    def _validate_inferred_outputs(self, output_meta_dict: dict, output_dict: dict):
+        """Validate inferred output dict against annotation."""
+        error_prefix = "Unmatched outputs between actual pipeline output and output in annotation"
+        if output_meta_dict.keys() != self.output_annotation.keys():
```

Comment:
Note: We can't get the 'mode' for registered component output, this may cause unexpected validation error.